public/JavaScript/gallery.js

```diff
@@ -62,4 +62,44 @@ function reportGalleryStats (imagesInDom) {
   return String(imagesInDom)
 }
 
-module.exports = reportGalleryStats
+/**
+ *
+ * @param {Array} userSelectedFilters
+ * @returns {Object} activeFilters
+ */
+function generateFilterOptions (userSelectedFilters) {
+  const allowedFilters = {
+    gender: ['male', 'female'],
+    type: ['dog', 'cat', 'fox', 'parrot'],
+    rating: [1, 2, 3, 4, 5]
+  }
+
+  const activeFilters = {
+    gender: null,
+    type: null,
+    rating: null
+  }
+
+  Object.keys(userSelectedFilters).forEach(filterKey => {
+    //  If the filter selected is not allowed then skip the process and go to the next
+    if (!Object.prototype.hasOwnProperty.call(allowedFilters, filterKey)) {
+      return
+    }
+    // If filter option is not in the allowed array set it too null
+    const filterValues = userSelectedFilters[filterKey]
+    if (!Array.isArray(filterValues)) {
+      activeFilters[filterKey] = null
```

Comment:
Why? 
It is initialised with null.
Is it changed during the execution of this function?

Reply:
Right they were already declared null, that line is useless there